src/Compilers/CSharp/Portable/Symbols/Source/SourceDelegateMethodSymbol.cs

```diff
@@ -27,6 +27,11 @@ internal abstract class SourceDelegateMethodSymbol : SourceMethodSymbol
         {
             _returnType = returnType;
             this.MakeFlags(methodKind, declarationModifiers, _returnType.SpecialType == SpecialType.System_Void, isExtensionMethod: false);
+
+            if (this.ReturnsByRefReadonly)
```

Comment:
>if (this.ReturnsByRefReadonly) [](start = 12, length = 30)

I think it would be better to make this part of completion for the symbol rather than eagerly do this in a constructor. We might get into some circularity where an attempt to look for the type will get us here again.